cmd/rpcdaemon/commands/eth_accounts.go

```diff
@@ -44,17 +45,34 @@ func (api *APIImpl) GetTransactionCount(ctx context.Context, address common.Addr
 		return nil, fmt.Errorf("getTransactionCount cannot open tx: %v", err1)
 	}
 	defer tx.Rollback()
-	blockNumber, _, err := rpchelper.GetBlockNumber(blockNrOrHash, tx, api.filters)
-	if err != nil {
-		return nil, err
+	var reader state.StateReader
+	var pending bool
+	if num, ok := blockNrOrHash.Number(); ok {
+		switch num {
+		case rpc.LatestBlockNumber:
+			reader = state.NewPlainStateReader(tx)
+		case rpc.PendingBlockNumber:
+			reader = state.NewPlainStateReader(tx)
+			pending = true
+		}
```

Comment:
is it 
- txPool.Nonce() method?
- or api.pendingBlock().Nonce()?
- or ?

Reply:
It is simply returning the nonce which is +1 from the nonce in the current state

Reply:
Actually, you are right, we need support from tx pool for that